### PR TITLE
Optimize the performance of the typed array pool on the native.

### PR DESCRIPTION
**Pull request by caryliu1999**
https://github.com/cocos-creator/3d-tasks/issues/6760

### CODE REVIEW

File: cocos/bindings/dop/BufferAllocator.cpp [hunk 1]
```diff
@@ -43,7 +43,7 @@ BufferAllocator::~BufferAllocator() {
         buffer.second->decRef();
     }
     _buffers.clear();
-    _caches.clear();
+    _array.clear();
```
Comment (minggo) [inline cocos/bindings/dop/BufferAllocator.cpp:46]: Do not have to clear them in destructor.

File: cocos/renderer/pipeline/deferred/DeferredPipeline.cpp [hunk 2]
```diff
@@ -128,7 +128,6 @@ bool DeferredPipeline::activate() {
 }
 
 void DeferredPipeline::render(const vector<uint> &cameras) {
-    if (cameras.empty()) return;
```
Comment (minggo) [inline cocos/renderer/pipeline/deferred/DeferredPipeline.cpp:131]: Why remove it?

File: cocos/bindings/dop/BufferAllocator.h [hunk 3]
```diff
@@ -86,9 +84,10 @@ class CC_DLL BufferAllocator final : public cc::Object {
     static cc::vector<BufferAllocator *> pools;
     static constexpr uint                BUFFER_MASK = ~(1 << 30);
 
-    cc::map<uint, Object *>  _buffers;
-    cc::map<uint, uint8_t *> _caches;
-    PoolType                 _type = PoolType::UNKNOWN;
+    cc::map<uint, Object *> _buffers;
+    cc::vector<uint8_t *>   _array;
```
Comment (minggo) [inline cocos/bindings/dop/BufferAllocator.h:88]: Should use a more meaningful name, such as _cachedBufferDataPointers.

File: cocos/bindings/dop/BufferAllocator.h [hunk 4]
```diff
@@ -86,9 +84,10 @@ class CC_DLL BufferAllocator final : public cc::Object {
...
+    cc::vector<uint> _sizes;
```
Comment (minggo) [inline cocos/bindings/dop/BufferAllocator.h:89]: The same as above, may use _cachedBufferLengths.

File: cocos/bindings/dop/BufferAllocator.h (hunk 3 above)
Comment (caryliu1999) [reply cocos/bindings/dop/BufferAllocator.h:88]: > Should use a more meaningful name, such as _cachedBufferDataPointers.

ok

File: cocos/renderer/pipeline/deferred/DeferredPipeline.cpp (hunk 2 above)
Comment (caryliu1999) [reply cocos/renderer/pipeline/deferred/DeferredPipeline.cpp:131]: > Why remove it?

ZeQiang made a judgment at the JS layer, it is not needed here. And not calling flushCommands may cause multi-thread blocking.

**Review by minggo — APPROVED** [review]

**Pull request merged**